File: app/src/main/java/com/romellbolton/putmeon/SessionManager.java
package com.romellbolton.putmeon;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

import java.util.HashMap;

public class SessionManager {

    private static SharedPreferences pref;
    private static SharedPreferences.Editor editor;
    private static Context context;
    int PRIVATE_MODE = 0;
    private static String PREF_NAME;
    private static final String IS_LOGIN = "IsLoggedIn";
    private static final String CLIENT_ID = "8740928683fe4ab6be03091a875ac618";

    public SessionManager() {
    }

    public void createLoginSession(Context context, String apiUserName) {
        this.context = context;
        PREF_NAME = apiUserName;
        pref = context.getSharedPreferences(PREF_NAME, Context.MODE_MULTI_PROCESS | PRIVATE_MODE);
        editor = pref.edit();
        editor.clear();
        editor.putBoolean(IS_LOGIN, true);
        editor.putString("id", apiUserName);
        editor.commit();
    }

    public String getToken() {
        return PREF_NAME;
    }

    public HashMap<String, String> getUserDetails() {
        HashMap<String, String> user = new HashMap<String, String>();
        user.put("id", pref.getString("id", null));
        return user;
    }

    public String getClientId() {
        return CLIENT_ID;
    }

    public void checkLogin() {
        if (!this.isLoggedIn()) {
            Intent i = new Intent(context, LoginActivity.class);
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(i);
        }
    }

    public String getPrefName() {
        return PREF_NAME;
    }

    public boolean isLoggedIn() {
        return pref.getBoolean(IS_LOGIN, false);
    }

    public void logoutUser() {
        editor.clear();
        editor.commit();
        Intent i = new Intent(context, LoginActivity.class);
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        context.startActivity(i);
    }
}

File: app/src/main/java/com/romellbolton/putmeon/util/SessionManager.kt
package com.romellbolton.putmeon.util

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import com.romellbolton.putmeon.controller.LoginActivity
import java.util.*

class SessionManager {
    var PRIVATE_MODE = 0
    fun createLoginSession(context: Context, apiUserName: String?) {
        pref = context.getSharedPreferences(apiUserName, Context.MODE_MULTI_PROCESS or PRIVATE_MODE)
        editor = pref.edit()
        editor.clear()
        editor.putBoolean(IS_LOGIN, true)
        editor.putString("id", apiUserName)
        editor.commit()
    }

    val userDetails: HashMap<String, String>
        get() {
            val user = HashMap<String, String>()
            user["id"] = pref.getString("id", null)
            return user
        }

    fun checkLogin() {
        if (!isLoggedIn) {
            val i = Intent(context, LoginActivity::class.java)
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            i.flags = Intent.FLAG_ACTIVITY_NEW_TASK
            context!!.startActivity(i)
        }
    }

    val isLoggedIn: Boolean
        get() = pref.getBoolean(IS_LOGIN, false)

    fun logoutUser() {
        editor.clear()
        editor.commit()
        val i = Intent(context, LoginActivity::class.java)
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
        i.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        context!!.startActivity(i)
    }

    companion object {
        private lateinit var pref: SharedPreferences
        private lateinit var editor: SharedPreferences.Editor
        private val context: Context? = null
        private const val IS_LOGIN = "IsLoggedIn"
        const val clientId = "8740928683fe4ab6be03091a875ac618"
    }
}